include/syncmer_hash.hpp

```diff
@@ -0,0 +1,135 @@
+    * \param[in] window_size       The number of values in one window
+    * \param[in] subwindow_size    The number of values to be checked as subwindow.
+    * \param[in] seed              The seed to use.
+    * \throws std::invalid_argument if the window size is smaller than 1.
+    * \returns                     A range of converted elements.
+    */
+    constexpr auto operator()(size_t const subwindow_size, size_t const window_size, seed const seed) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, subwindow_size, window_size, seed};
+    }
+
+    /*!\brief Call the view's constructor with the underlying view, a window size and a subwindow size as argument.
+     * \param[in] urange            The input range to process. Must model std::ranges::viewable_range and
+     *                              the reference type of the range must model seqan3::semialphabet.
+     * \param[in] window_size       The number of values in one window
+     * \param[in] subwindow_size    The number of values to be checked as subwindow.
+     * \param[in] seed              The seed to use.
+     * \throws std::invalid_argument if the subwindow size is smaller than 1 or window_size is smaller than subwindow_size.
+     * \returns                     A range of converted elements.
+     */
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange,
+                              size_t const subwindow_size,
+			       size_t const window_size,
+                              seed const seed = seqan3::seed{0x8F3F73B5CF1C9ADE}) const
+    {
+        static_assert(std::ranges::viewable_range<urng_t>,
+            "The range parameter to views::syncmer_hash cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::forward_range<urng_t>,
+            "The range parameter to views::syncmer_hash must model std::ranges::forward_range.");
+        static_assert(semialphabet<std::ranges::range_reference_t<urng_t>>,
+            "The range parameter to views::syncmer_hash must be over elements of seqan3::semialphabet.");
+
+        if (subwindow_size < 1 || window_size < subwindow_size)
+            throw std::invalid_argument{"The chosen window_size and subwindow_size are not valid."
+                                        "Please choose values greater than 1 and a subwindow_size smaller than the window_size."};
+
+        auto window_hash = std::forward<urng_t>(urange) | seqan3::views::kmer_hash(seqan3::shape(seqan3::ungapped(window_size))) 
+                                                        | std::views::transform([seed](uint64_t i){return i ^ seed.get();});
+
+        auto subwindow_hash = std::forward<urng_t>(urange) | seqan3::views::kmer_hash(seqan3::shape(seqan3::ungapped(subwindow_size)))
+                                                       | std::views::transform([seed](uint64_t i){return i ^ seed.get();});
+
```

Comment:
I think, for syncmer_hash we either offer the option to use both strands (preferable) or use both strands as defaults. You can look at the modmer_hash function for how to do that.